### PR TITLE
vim-patch:8.1.0941: macros for MS-Windows are inconsistent

### PR DESCRIPTION
**Pull request by dundargoc**
####  vim-patch:8.1.0941: macros for MS-Windows are inconsistent

Problem:    Macros for MS-Windows are inconsistent, using "32", "3264 and
            others.
Solution:   Use MSWIN for all MS-Windows builds.  Use FEAT_GUI_MSWIN for the
            GUI build. (Hirohito Higashi, closes vim/vim#3932)
https://github.com/vim/vim/commit/4f97475d326c2773a78561fb874e4f23c25cbcd9

### CODE REVIEW

**Comment by zeertzjq** [comment]
How about replacing WIN32 with MSWIN instead?

**Comment by zeertzjq** [comment]
https://github.com/vim/vim/commit/4f97475d326c2773a78561fb874e4f23c25cbcd9

File: test/functional/fixtures/tty-test.c [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <uv.h>
-#ifdef _WIN32
+#ifdef _MSWIN
```
Comment (zeertzjq) [inline test/functional/fixtures/tty-test.c:8]: Is `_MSWIN` defined?

Comment (dundargoc) [reply test/functional/fixtures/tty-test.c:8]: Doh, it's not. Thanks.

**Review by justinmk — APPROVED** [review]
I don't think it's inconsistent in nvim codebase, but no reason to diverge here.

File: CMakeLists.txt [hunk 2]
```diff
@@ -273,7 +273,7 @@ if(MSVC)
   # XXX: /W4 gives too many warnings. #3241
   add_compile_options(/W3)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE)
-  add_definitions(-DWIN32)
+  add_definitions(-DMSWIN)
```
Comment (zeertzjq) [inline CMakeLists.txt:276]: Also defined this for MinGW?

Comment (dundargoc) [reply CMakeLists.txt:276]: Hmmm, no that doesn't sound right. I think this is only relevant for the MSVC compiler.

Comment (dundargoc) [reply CMakeLists.txt:276]: Hmm, I'll need to test with MINGW to be sure though.

Comment (dundargoc) [reply CMakeLists.txt:276]: Nope, you're right. Thanks for the help!

**Pull request merged**